### ISSUE TITLE
Use <details> instead of roll-your-own content hiding

### ISSUE THREAD

**Issue by fluffy-critter** [tootsuite/mastodon#12212]
### Pitch

Update the web UI HTML to use `<details>` instead of `<button>`, for better markup semantics.

This is similar to #1464

### Motivation

This would improve accessibility, particularly for users of screen readers.

**Comment by jalcine** [comment]
Even using a `:focus` approach could work here!

**Comment by ZoeBijl** [comment]
Related to the `:focus` part: https://github.com/tootsuite/mastodon/issues/12275

**Comment by nightpool** [comment]
@fluffy-critter Can you (or someone else) update this issue with the actual, practical accessibility benefits using `<details>` would provide here? Just saying "This would improve accessibility" is not enough justification—what accessibility goals or success criteria are we attempting to fulfill, or currently don't meet? (aside from the Read More :focus issue, which is unrelated and should be tracked separately)

In particular, as far as trade-offs go, is it possible to keep details collapsed/expanded state in sync across multiple different DOM elements, the way our current content warnings work? (If a status's content warning is expanded, then it is expanded in all views a status can be shown in, and similarly when it is collapsed)

**Comment by nightpool** [comment]
(The :focus issue is unrelated, it's an unintentional bug in the CSS code for statuses, caused by weird selector priority, and would still be the case using `<details>` or any other element)

**Comment by fluffy-critter** [comment]
Expanding/collapsing `<details>` elements can be accomplished by setting/clearing the `.open` attribute on them. See the snippets at https://stackoverflow.com/a/43009778/318857 for examples.

As far as the specific accessibility concern, using a semantic tag means that a screen reader has a better chance of understanding what the tag is and providing appropriate voice narration accordingly. It's good to be proactive about supporting these things so that the greater ecosystem gets a positive impact in the long term; breaking the chicken-and-egg accessibility cycle is important.

**Comment by bkil** [comment]
Related: #17115

**Comment by csarven** [comment]
I also find the `details` interactive element to be most appropriate for the "Show more" (and "Show less") feature in the UI. `details` has native keyboard/mouse support in Web browsers, doesn't require JavaScript, provides styling possibilities, and ARIA.

HTML's [details element](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-details-element) notes:

>Visually and programmatically grouping related elements together can be important for accessible user experiences. This can help users understand the relationship between such elements. When related elements are in disparate sections of a web page rather than being grouped, the elements' relationships to each other can be less discoverable or understandable.

Which seems to be appropriate for the show/more case.

Moreover, ARIA and ARIA in HTML both acknowledge the role of `aria-details` in markup languages (which corresponds to native `details` in HTML) for accessible documents, e.g., see [aria-details](https://www.w3.org/TR/wai-aria/#aria-details), [details](https://www.w3.org/TR/html-aria/#ref-for-dfn-no-role-10).

---

Something along the following lines could work (with the intention to retain the current UI). The HTML is intentionally verbose and indented for readability here:

```html
<details>
  <summary>Show
    <span class="show-more">more</span>
    <span class="show-slash">/</span>
    <span class="show-less">less</span>
  </summary>
  💩 
</details>
```

The idea is to use CSS along these lines:

```css
details summary .show-slash,
details:not([open]) summary .show-less,
details[open] summary .show-more {
display:none;
}
```

Without the CSS (works on GitHub's HTML rendering of this issue at the time of this writing):

<details>
  <summary>Show <span class="show-more">more</span><span class="show-slash">/</span><span class="show-less">less</span></summary>
  💩 
</details>
